storage/database/src/main/java/de/danoeh/antennapod/storage/database/PodDBAdapter.java

```diff
@@ -1113,18 +1113,22 @@ public Cursor getPlayedItemsCursor() {
      *              integer >= 0.
      * @throws IllegalArgumentException if limit < 0
      */
-    public final Cursor getCompletedMediaCursor(int offset, int limit) {
+    public final Cursor getCompletedMediaCursor(int offset, int limit, long[] timestamps) {
         if (limit < 0) {
             throw new IllegalArgumentException("Limit must be >= 0");
         }
+        String query = "SELECT * FROM " + TABLE_NAME_FEED_MEDIA
+                + " WHERE " + KEY_PLAYBACK_COMPLETION_DATE + " > " + timestamps[0]
+                + " AND " + KEY_PLAYBACK_COMPLETION_DATE + " <= " + timestamps[1]
+                + " ORDER BY " + String.format(Locale.US, "%s DESC LIMIT %d, %d",
+                KEY_PLAYBACK_COMPLETION_DATE, offset, limit);
```

Comment:
I know this was there before, but I don't really see a reason for using expensive `String.format` calls here. While already touching this part of the code, could you please simplify this to something like this?

```suggestion
                + " ORDER BY " + KEY_PLAYBACK_COMPLETION_DATE + " DESC"
                + " LIMIT " + offset + ", " + limit;
```